cpp-client/deephaven/client/include/public/deephaven/client/client.h

```diff
@@ -1751,4 +1762,576 @@ TableHandle TableHandle::ungroup(bool nullFill, Args &&... columnSpecs) const {
   };
   return ungroup(nullFill, std::move(groupByColumns));
 }
+
```

Comment:
Would it make sense to have all this UpdateBy stuff (the class and the free standing functions and typedefs after it) on its own header file, as opposed to riding along on client.h?  Not sure if client.h could do with just a forward declare or not, but if it can, then case (a) win! and if it cannot, then case (b) at least we have some code separation in smaller pieces?

Reply:
Done. In this case client.h just forward-declares. If customers want to use updateBy, they will have to include update_by.h

(We could also have client.h include update_by.h if you prefer)

Reply:
I think this is perfect.